Macros IS_GUEST_KEY

MNarath1 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This PR simply macros an old Proc that checks if a key is a guest key or not by simply making it a string comparison maybe still not the best solution but better than the original approach definitly.
Now this also makes the macro with 2 inputs for a very simple reason the macro is used in wildly different enviroments most do have a client datum but not all of them so i opted to simply get the var here directly and individually to the usecase.
## Why It's Good For The Game
one global proc less
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
Also put closed issues under this tag, if any. Format is as follows(must be lowercase):
closes #123456789
-->
</details>

## Changelog
:cl:
refactor: Refactored guest player detection to be much simpler.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

MNarath1 committed:
nothing yet

MNarath1 committed:
better

MNarath1 committed:
small fix and changes

MNarath1 committed:
two lines escaped whops

MNarath1 committed:
small Comment change

ike709 commented:
This *seems* fine but it could still be sus. I kind of wonder how this affects CID spoofers.

MNarath1 commented:
> This _seems_ fine but it could still be sus. I kind of wonder how this affects CID spoofers.

considering that key's probably are send to the server (not perfectly sure on that) this may be a problem i mean it would not be much of an issue if the key is generated server side but a big one if its client side cause then someone just needs to change either their cid afterwards for this to not work anymore

ike709 commented on code/__DEFINES/is_helpers.dm:
```diff
@@ -11,6 +11,9 @@
 #define isweakref(D) (istype(D, /datum/weakref))
 
+// simple check whether or not a player is a guest using their key and computer_id to build the key and then make a string comparison with the actual key
+#define IS_GUEST_KEY(key, computer_id)	("[key]" == "Guest-[computer_id]")
```
After pondering it, to stay on the safe side I think you should check if it starts with `Guest-` and then if the remainder of it is a number.

ike709 replied:
Or just only check that it starts with `Guest-`

ike709 replied:
Yeah just check if it starts with `Guest-`

MNarath1 committed:
lets just check for Guest-

MNarath1 committed:
using findtext as its a bit more cleaner

both methodes are pretty much just as fast but this seemed like the cleaner option

ike709 commented on code/__DEFINES/is_helpers.dm:
```diff
@@ -11,6 +11,9 @@
 #define isweakref(D) (istype(D, /datum/weakref))
 
+// simple check whether or not a player is a guest using their key
+#define IS_GUEST_KEY(key)	(findtext(key, "Guest-", 1, 7))
```
```suggestion
#define IS_GUEST_KEY(key)	(findtextEx(key, "Guest-", 1, 7))
```

MNarath1 committed:
apply suggestions
